### PR TITLE
(PUP-9472) Emit X509::Name as UTF-8

### PR DESCRIPTION
**Pull request by joshcooper**
The `OpenSSL::X509::Name#to_s` method uses the default "oneline"
format, which corrupts non-ASCII characters. For example, the 𠜎 character when
UTF-8 encoded is 0xF0, 0xA0, 0x9C, 0x8E, but `Name#to_s` encodes each byte as
the 4 byte sequence `\\`, 'x', &lt;hex&gt;, &lt;hex&gt;, resulting in a 16 byte sequence.

    irb(main):006:0> name = "CN=root-ca-𠜎"
    => "CN=root-ca-𠜎"
    irb(main):007:0> OpenSSL::X509::Name.parse(name).to_s
    => "/CN=root-ca-\\xF0\\xA0\\x9C\\x8E"
    irb(main):008:0> OpenSSL::X509::Name.parse(name).to_utf8
    => "CN=root-ca-𠜎"

Use `OpenSSL::X509::Name#to_utf8` explicitly. This is mostly an issue
for CA certs, since the subject doesn't correspond to a hostname and can
be anything like:

    C=HU, L=Budapest, O=NetLock Kft., OU=Tanúsítványkiadók (Certification Services), CN=NetLock Arany (Class Gold) Főtanúsítvány

### CODE REVIEW

**Comment by joshcooper** [comment]
Need to retarget 6.4.x and update mocha/rspec conflicts

File: spec/unit/util/monkey_patches_spec.rb [hunk 1]
```diff
@@ -106,7 +106,7 @@ def with_root_certs(certs)
       with_root_certs([cert])
       store.add_cert(cert)
 
-      expect(store).to receive(:warn).with('Failed to add /DC=com/DC=microsoft/CN=Microsoft Root Certificate Authority')
+      expect(store).to receive(:warn).with('Failed to add CN=Microsoft Root Certificate Authority,DC=microsoft,DC=com')
```
Comment (joshcooper) [inline spec/unit/util/monkey_patches_spec.rb]: Note if the distinguished name has multiple components, then the order is reversed from what it was prior. I think the new behavior is more correct/expected.

**Comment by joshcooper** [comment]
Submitted a PR against master for the cruft: https://github.com/joshcooper/puppet/pull/new/delete_cruft

**Comment by Iristyle** [comment]
Oh for the love of...

**Pull request merged**